translate/src/context/EntitiesList.tsx

```diff
@@ -0,0 +1,27 @@
+import { createContext, useState } from 'react';
+
+type EntitiesList = Readonly<{
+  visible: boolean;
+  show(visible: boolean): void;
+}>;
+
+const initEntitiesList: EntitiesList = {
+  visible: false,
+  show: () => {},
+};
+
+export const EntitiesList = createContext(initEntitiesList);
+
+export function EntitiesListProvider({
+  children,
+}: {
+  children: React.ReactElement;
+}) {
+  const [state, setState] = useState(() => ({
+    ...initEntitiesList,
+    show: (visible: boolean) => setState((prev) => ({ ...prev, visible })),
+  })) as [EntitiesList, React.Dispatch<React.SetStateAction<EntitiesList>>];
```

Comment:
Shouldn't need a typecast here.

```suggestion
  const [state, setState] = useState<EntitiesList>(() => ({
    ...initEntitiesList,
    show: (visible: boolean) => setState((prev) => ({ ...prev, visible })),
  }));
```